### PR TITLE
fix(ingest/dbt): ensure dbt shows view properties

### PR DESCRIPTION
**Pull request by hsheth2**
Fixes https://github.com/datahub-project/datahub/issues/7868.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/integration/dbt/dbt_test_column_meta_mapping_golden.json [hunk 1]
```diff
@@ -129,7 +130,8 @@
         "json": {
             "typeNames": [
                 "Model",
-                "Table"
+                "Table",
```
Comment (jjoyce0510) [inline metadata-ingestion/tests/integration/dbt/dbt_test_column_meta_mapping_golden.json:133]: Something can have all these different subtypes at once?? Whoa.. What's expected in the UI?

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py [hunk 2]
```diff
@@ -1352,7 +1352,9 @@ def _create_subType_wu(
         subtypes: List[str] = [node.node_type.capitalize()]
         if node.materialization == "table":
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py:1353]: minor nit: can these typenames go into an enum?

File: metadata-ingestion/tests/integration/dbt/dbt_test_column_meta_mapping_golden.json (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/tests/integration/dbt/dbt_test_column_meta_mapping_golden.json:133]: Yep - the UI shows the first one, but takes all of them into account when deciding whether or not to show the "View Properties" tab

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py:1353]: Ideally yes, but that's a bigger refactor - don't want to do that now

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**